### ISSUE TITLE
getCsvDocument() - public or protected?

### ISSUE THREAD

**Issue by drjonnicholson** [thephpleague/csv#216]
## Issue summary

Question: Is `getCsvDocument()` supposed to be `public` or `protected`?

We want to get the file/stream of a `Writer`, which is marked as `protected` in `AbstractCsv`.  `AbstractCsv` uses the `ControlsTrait`. `ControlsTrait` has `getCsvDocument()` as `abstract public`.

### System informations

N/A

## Standalone code, or other way to reproduce the problem

N/A

### Expected result

```php
$writer = new Writer(new \SplTempFileObject());
// sometime later...
$writer->getCsvDocument(); // gets the temporary file's path
```

### Actual result

```php
$writer = new Writer(new \SplTempFileObject());
// sometime later...
$writer->getCsvDocument(); // Error: Call to undefined method League\Csv\Writer::getCsvDocument()
```

**Comment by nyamsprod** [comment]
I believe there is no geCsvDocument in any stable version of the package. The master which is stable should not be use in production has it expose an intermediary API to enable working on the next major stable version. 
TL;DR: getCSvDocument will be remove in the final v9 API and does not exist in v8 AFAIK

**Comment by drjonnicholson** [comment]
Ahh, yes different version of code to the package installed. Didn't think to check!

Quite right about the SplTempFileObject, and we just resolved our issue in another way anyway making our issue moot. Thanks for the feedback, closing the issue.
